go/felix/intdataplane/ipip_mgr_netlink.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intdataplane
+
+import (
+	. "github.com/vishvananda/netlink"
+	"os/exec"
+)
+
+// ipipDataplane is a shim interface for mocking netlink and os/exec in the IPIP manager.
+type ipipDataplane interface {
+	LinkByName(name string) (Link, error)
+	LinkSetMTU(link Link, mtu int) error
+	LinkSetUp(link Link) error
+	AddrList(link Link, family int) ([]Addr, error)
+	AddrAdd(link Link, addr *Addr) error
+	AddrDel(link Link, addr *Addr) error
+	RunCmd(name string, args ...string) error
```

Comment:
I wondered that too, but doing that would put more code in the shim, where it's not possible to UT